modules/auxiliary/gather/qnap_backtrace_admin_hash.rb

```diff
@@ -90,11 +90,10 @@ def run
 
     if admin_hash
       print_good("Hopefully this is your hash: #{admin_hash}")
-      report_note(
-        host: rhost,
-        port: rport,
-        type: 'qnap.admin.hash',
-        data: admin_hash
+      store_valid_credential(
```

Comment:
store_vaild_credential is expected to cover the 80% case for modules, service only creds look to have different expectations. Maybe something like

```
    connection_details = {
          module_fullname: self.fullname,
          username: 'admin',
          private: admin_hash,
          private_type: :nonreplayable_hash,
          workspace_id: myworkspace_id,
      }.merge(service_details)
      create_credential(connection_details)
```

I doesn't save as much code however if does take advantage of the standard service details.